streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -274,6 +289,71 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
             thread.setUncaughtExceptionHandler(eh);
     }
 
+
+    /**
+     * Find all of the instances of {@link StreamsMetadata} in a {@link KafkaStreams application}
+     *
+     * Note: this is a point in time view and it may change due to partition reassignment.
+     * @return collection containing all instances of {@link StreamsMetadata} in this application
+     */
+    public Collection<StreamsMetadata> allMetadata() {
+        validateIsRunning();
+        return streamsMetadataState.getAllMetadata();
+    }
+
+
+    /**
+     * Find the instances {@link StreamsMetadata} for a given storeName
```

Comment:
nit: `Find instances of {@link StreamsMetadata} that contains the given storeName.